### DISCUSSION TITLE
exceptions from base package not handle on response handler

### DISCUSSION THREAD

**Discussion by Artessan** [creagia/laravel-redsys#78] [Q&A]
The HandleRedsysResponse.php file deals only with DeniedRedsysPaymentResponseException. 

ErrorRedsysResponseException and InvalidRedsysResponseException are not taken care of. 

It is done purposely or it is a flaw? . It could be beneficial to handle responses such as "Cancelado por el titular" that as part of ErrorRedsysResponseException only result in the exception being raised.

**Comment by dtorras** [comment]
Hello! 
For `InvalidRedsysResponseException` it's intended **to not catch** the exception because we won't be able to match any order or more information. It's very unlikely to receive a post from Redsys that throws this exception.

With `ErrorRedsysResponseException` and responses like "Cancelado por el titular" I would need to investigate it further because it seems interesting to handle. Do you have a response example with error code? Can we match the payment request and change the status in this case?
